Mathlib/CategoryTheory/IsConnected.lean

```diff
@@ -331,6 +331,14 @@ theorem zigzag_obj_of_zigzag (F : J ⥤ K) {j₁ j₂ : J} (h : Zigzag j₁ j₂
     Zigzag (F.obj j₁) (F.obj j₂) :=
   h.lift _ fun _ _ => Or.imp (Nonempty.map fun f => F.map f) (Nonempty.map fun f => F.map f)
 
+/-- A zigzag in a discrete category entails an equality of its extremities -/
+lemma eq_of_zigzag (X) {a b : Discrete X} (zab : Relation.ReflTransGen Zag a b) : a.as = b.as := by
+  induction zab with
+  | refl => rfl
+  | @tail b c _ zbc eqab  =>
+    exact eqab.trans ( zbc.elim (Nonempty.elim · Discrete.eq_of_hom)
+      (Eq.symm ∘ (Nonempty.elim · Discrete.eq_of_hom)))
```

Comment:
```suggestion
lemma eq_of_zigzag (X) {a b : Discrete X} (h : Zigzag a b) : a.as = b.as := by
  induction h with
  | refl => rfl
  | tail _ h eq  =>
    exact eq.trans (h.elim (fun ⟨f⟩ ↦ Discrete.eq_of_hom f) 
      (fun ⟨f⟩ ↦ (Discrete.eq_of_hom f).symm))
```

I always try to make variable names consistent throughout the file. I changed the proof in the `tail` case slightly, to me this is more readable. Maybe it's worth proving a lemma `eq_of_zag` as well, and apply that in the inductive proof.